manager/manager/manager.go

```diff
@@ -104,21 +104,40 @@ func (m *manager) CreateDeployment(t *Template) (*Deployment, error) {
 		return nil, err
 	}
 
-	err = m.repository.AddManifest(t.Name, manifest)
+	actualConfig, err := m.deployer.CreateConfiguration(manifest.ExpandedConfig)
+	log.Printf("Got Back %s", actualConfig)
 	if err != nil {
-		log.Printf("AddManifest failed %v", err)
+		// Deployment failed, mark as failed
+		log.Printf("CreateConfiguration failed: %v", err)
 		m.repository.SetDeploymentStatus(t.Name, FailedStatus)
-		return nil, err
+		// If we failed before being able to create some of the resources, then
+		// return the failure as such. Otherwise, we're going to add the manifest
+		// and hence resource specific errors down below.
+		if actualConfig == nil {
+			return nil, err
+		}
+	} else {
+		m.repository.SetDeploymentStatus(t.Name, DeployedStatus)
 	}
-
-	if err := m.deployer.CreateConfiguration(manifest.ExpandedConfig); err != nil {
-		// Deployment failed, mark as deleted
-		log.Printf("CreateConfiguration failed: %v", err)
+	
+	// Update the manifest with the actual state of the reified resources
+	manifest.ExpandedConfig = actualConfig
+	aErr := m.repository.AddManifest(t.Name, manifest)
```

Comment:
This gets a little funky. It's fine for now because we're synchronous and probably only have one person working at a time, but eventually we really need to add the manifest first and modify it as things change (i.e., in a potentially long-running deployment, I want to see the manifest that's being deployed, and the live status of the operation). This change breaks that model by not creating a manifest resource until resourcifier is done.